Census/Listeners/CensusListener.cs

```diff
@@ -57,67 +59,96 @@ private async System.Threading.Tasks.Task StartConsumerLoop(CancellationToken ca
         kafkaConsumer.Subscribe(KafkaTopic.PatientIDsAcquired.ToString());
         ConsumeResult<string, string> rawmessage = null;
 
-        while (!cancellationToken.IsCancellationRequested)
+        try
         {
-            rawmessage = kafkaConsumer.Consume();
-            IBaseMessage deserializedMessage = null;
-            (string facilityId, string correlationId) messageMetaData = (string.Empty, string.Empty);
-
-            if (rawmessage != null)
+            while (!cancellationToken.IsCancellationRequested)
             {
-                IBaseMessage message = null;
                 try
                 {
-                    message = DeserializeMessage(rawmessage.Topic, rawmessage.Message.Value);
+                    rawmessage = kafkaConsumer.Consume(cancellationToken);
                 }
-                catch (Exception ex)
+                catch(ConsumeException ex)
                 {
-                    var errorMessage = $"Unable to deserialize message: {rawmessage?.Message?.Value}";
-                    _logger.LogError(errorMessage);
+                    _nonTransientExceptionHandler.HandleException(ex);
+                    _logger.LogError($"Error consuming message: {ex.Error.Reason}");
                     continue;
                 }
 
-                deserializedMessage = message;
-                messageMetaData = ExtractFacilityIdAndCorrelationIdFromMessage(rawmessage.Message);
+                IBaseMessage deserializedMessage = null;
+                (string facilityId, string correlationId) messageMetaData = (string.Empty, string.Empty);
 
-                if (string.IsNullOrWhiteSpace(messageMetaData.facilityId))
+                if (rawmessage != null)
                 {
-                    var errorMessage = $"No Facility ID provided. Unable to process message: {message}";
-                    _logger.LogWarning(errorMessage);
-                    continue;
-                }
+                    IBaseMessage message = null;
+                    try
+                    {
+                        message = DeserializeMessage(rawmessage.Topic, rawmessage.Message.Value);
+                    }
+                    catch (Exception ex)
+                    {
+                        _nonTransientExceptionHandler.HandleException(rawmessage,ex);
+                        var errorMessage = $"Unable to deserialize message: {rawmessage?.Message?.Value}";
+                        _logger.LogError(errorMessage);
```

Comment:
Hey Ed, the event needs to be committed here after the nontransient handles the exception

Reply:
added